egcg_core/notifications/asana.py

```diff
@@ -15,10 +16,23 @@ def __init__(self, name, workspace_id, project_id, access_token, task_descriptio
         if task_description:
             self.task_template['notes'] = task_description
 
-    def notify(self, msg):
+    def notify(self, msg, attachments=None):
         self.client.tasks.add_comment(self.task['id'], text=msg)
         self.client.tasks.update(self.task['id'], completed=False)
 
+        if attachments:
+            if isinstance(attachments, str):
+                attachments = [attachments]
+
+        for attachment in attachments or []:
+            with open(attachment, "rb") as fil:
+                content = fil.read()
+                self.client.attachments.create_on_task(
+                    task_id=self.task['id'],
+                    file_content=content,
+                    file_name=basename(attachment)
+                )
```

Comment:
If we send a message to Asana again with the same file name, this attachment will be duplicated in the Asana task multiple times. Not sure if this would be a problem, but worth bearing in mind.